src/js/Features.js

```diff
@@ -162,78 +162,71 @@ Features.prototype.generateElements = function (featuresElements) {
-                    + feature_tip_html + '</td></tr>');
-
-            var feature_e = newElement.find('input.feature');
+            const featureElem = newElement.find('input.feature');
 
-            feature_e.prop('checked', bit_check(self._featureMask, self._features[i].bit));
-            feature_e.data('bit', self._features[i].bit);
+            featureElem.prop('checked', bit_check(self._featureMask, self._features[i].bit));
+            featureElem.data('bit', self._features[i].bit);
 
             newElements.push(newElement);
         }
 
         featuresElements.each(function () {
-            if ($(this).hasClass(self._features[i].group)) {
+            if ($(this).hasClass(self._features[i]?.group)) {
                 $(this).append(newElements);
             }
         });
     }
 
-    for (var i = 0; i < listElements.length; i++) {
-        var element = listElements[i];
-        var bit = parseInt(element.attr('value'));
-        var state = bit_check(self._featureMask, bit);
+    for (const element of listElements) {
+        const bit = parseInt(element.attr('value'));
+        const state = bit_check(self._featureMask, bit);
 
         element.prop('selected', state);
     }
 };
 
 Features.prototype.findFeatureByBit = function (bit) {
-    var self = this;
+    const self = this;
 
-    for (var i = 0; i < self._features.length; i++) {
-        if (self._features[i].bit == bit) {
+    for (const i in self._features) {
+        if (self._features[i].bit === bit) {
             return self._features[i];
         }
     }
 }
 
 Features.prototype.updateData = function (featureElement) {
-    var self = this;
+    const self = this;
 
     if (featureElement.attr('type') === 'checkbox') {
-        var bit = featureElement.data('bit');
+        const bit = featureElement.data('bit');
+        let featureValue;
 
-        var featureValue;
         if (featureElement.is(':checked')) {
             self._featureMask = bit_set(self._featureMask, bit);
             featureValue = 'On';
         } else {
             self._featureMask = bit_clear(self._featureMask, bit);
             featureValue = 'Off';
         }
-        self._analyticsChanges['Feature' + self.findFeatureByBit(bit).name] = featureValue;
+        self._analyticsChanges[`Feature${self.findFeatureByBit(bit).name}`] = featureValue;
     } else if (featureElement.prop('localName') === 'select') {
-        var controlElements = featureElement.children();
-        var selectedBit = featureElement.val();
+        const controlElements = featureElement.children();
+        const selectedBit = featureElement.val();
         if (selectedBit !== -1) {
-            var selectedFeature;
-            for (var i = 0; i < controlElements.length; i++) {
-                var bit = controlElements[i].value;
+            let selectedFeature;
+            for (const i in controlElements) {
```

Comment:
Still not super happy with doing it this way - if somebody was to add something like `controlElements.locked = true` this will break as `in` does **not** enumerate the elements of a structured datat type, but the properties.